### PR TITLE
[MLMD] Upstreamed all KFP Lineage changes till 2020-03-06.

### LINKED ISSUE #149: [MLMD Lineage] Implement routing / deep linking to Artifact Details Tabs
/kind feature
/area front-end
/area metadata
/assign @kwasi
/cc @avdaredevil

### PR DESCRIPTION
**Pull request by avdaredevil**
_**Addresses: #217**_
***fixes #149***

## About
- Upstreamed all KFP Lineage changes till 2020-03-06
- Changes:
  - Router Updates
  - Stripping out a ton of pipelines, runs and archives code from updates
  - Deep linking support added with a combination of HashRouter patch
  - Execution and Artifact details page now includes all the key-value pairs from metadata, with the new table layout

### Meta
/area metadata
/area front-end
/priority p1
/assign @avdaredevil
/cc @kwasi @Bobgy @jlewi 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/218)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -31,7 +32,9 @@
     "enzyme-to-json": "^3.3.5",
     "eslint-plugin-import": "^2.18.2",
     "license-checker": "^25.0.1",
+    "prettier": "^1.19.1",
```
Comment (Bobgy) [inline frontend/package.json]: You should pin prettier to exact version as KFP, because there can be formatting change between minor versions too.

Comment (avdaredevil) [reply frontend/package.json]: Done!

File: frontend/src/components/SideNav.tsx [hunk 2]
```diff
@@ -0,0 +1,429 @@
+/*
```
Comment (Bobgy) [inline frontend/src/components/SideNav.tsx]: Does metadata UI need this component?

Comment (avdaredevil) [reply frontend/src/components/SideNav.tsx]: It's used in the router component, and I don't really know how to remove it from there

Comment (Bobgy) [reply frontend/src/components/SideNav.tsx]: So in metadata UI, router is not rendered, is that right? Then it's fine

Comment (avdaredevil) [reply frontend/src/components/SideNav.tsx]: That's correct!

**Comment by zhenghuiwang** [comment]
LGTM. Thanks for making the changes. Any plan to componentize the code so that we don't need a manual merge them?

 presubmit tests should be passed once you rebase.

**Comment by Bobgy** [comment]
@zhenghuiwang Current problem is we don't have enough bandwidth (and priority) on componentizing all these.
I will try refactoring some as I develop major new features at best effort.

**Comment by zhenghuiwang** [comment]
> @zhenghuiwang Current problem is we don't have enough bandwidth (and priority) on componentizing all these.
> I will try refactoring some as I develop major new features at best effort.

Thanks @Bobgy

**Comment by avdaredevil** [comment]
Attn: @jlewi 
## From Meeting Notes:
- This PR is stuck because there are tests failing:
  - Some of these tests were added in Metadata and don't exist in Pipelines
  - As there is a lot of code complexity from Pipelines and no benchmark to base these tests, they will either need to be removed or updated manually (for which I don't have any cycles)
- Ownership?
  - Most code here is from KFP, with notable contributions from Jason, AP, Kwasi, etc.
  - Since the original codebase is forked off KFP, we'd like to:
    - Migrate the Details pages and Routing logic to kubeflow/frontend (for shareability)
      - Important for customers to provide a platform agnostic layer that shows metadata from all sources
    - OR, load the UI off KFP OSS
      - Easier, but makes Kubeflow opinionated, and may be a struggle as new metadata queries may come to KFP directly rather than being addressed as PRs from the community

**Comment by jlewi** [comment]
@zhenghuiwang @neuromage any suggestions on the path forward here?

**Comment by zhenghuiwang** [comment]
Thanks @avdaredevil for the summarization of the situation and possible solutions.

Abstracting out the shared logic/components looks right to me. If there isn't much difference between kubeflow metadata UI and Pipeline UI then abstracting out shouldn't require much effort. If KFP metadata UI is very specific to KFP, then loading it directly seems not a proper solution. Having the reusable abstraction will reduce the work of integrating it to these two UIs.
  
> * Ownership?
>   * Since the original codebase is forked off KFP, we'd like to:
>     
>     * Migrate the Details pages and Routing logic to kubeflow/frontend (for shareability)
>       
>       * Important for customers to provide a platform agnostic layer that shows metadata from all sources

**Comment by avdaredevil** [comment]
Ping @zhenghuiwang @Bobgy.

For now should we disable broken tests, mark as skip, and merge. (we can verify manually, since I tested this manually a while back)

**Comment by zhenghuiwang** [comment]
Sounds good to me. 
> For now should we disable broken tests, mark as skip, and merge. (we can verify manually, since I tested this manually a while back)

**Comment by Bobgy** [comment]
SGTM too

**Comment by jlewi** [comment]
@avdaredevil looks like you got the greenlight to disable tests to get this merged.

**Comment by jlewi** [comment]
@avdaredevil Any plans to update this and merge it?

**Comment by jlewi** [comment]
Merging manually just so we can close this out.

**Pull request merged**